File: app/src/main/java/com/example/myapplication/View/NotebookFragment.java
package com.example.myapplication.View;
import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.LifecycleOwner;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProvider;
import androidx.recyclerview.widget.RecyclerView;

import com.example.myapplication.Adapter.CatalogAdapter;
import com.example.myapplication.Other.VerticalSpacesItemDecoration;
import com.example.myapplication.ViewModel.NotebookViewModel;
import com.example.myapplication.databinding.FragmentNotebookBinding;

import java.util.ArrayList;
import java.util.Date;

import Models.linked.Product;
public class NotebookFragment extends Fragment implements LifecycleOwner {
    private FragmentNotebookBinding binding;
    private NotebookViewModel viewModel;
    RecyclerView NotebookRecView;
    CatalogAdapter adapter;
    public View onCreateView(@NonNull LayoutInflater inflater,
                             ViewGroup container, Bundle savedInstanceState) {
        binding = FragmentNotebookBinding.inflate(inflater, container, false);
        View root = binding.getRoot();
        NotebookRecView = binding.NotebookRecView;
        VerticalSpacesItemDecoration SpacesItemDecoration = new VerticalSpacesItemDecoration(15);
        NotebookRecView.addItemDecoration(SpacesItemDecoration);
        viewModel = new ViewModelProvider(this).get(NotebookViewModel.class);
        viewModel.getUserMutableLiveData().observe((LifecycleOwner) getContext(), userListUpdateObserver);
        return root;
    }
    Observer<ArrayList<Product>> userListUpdateObserver = new Observer<ArrayList<Product>>() {
        @Override
        public void onChanged(ArrayList<Product> userArrayList) {
            adapter = new CatalogAdapter(userArrayList);
            NotebookRecView.setAdapter(adapter);
            Date EndDate = new Date();
            Log.d("TIMER",String.valueOf(EndDate.getTime()));
        }
    };
    @Override
    public void onDestroyView() {
        super.onDestroyView();
        binding = null;
    }
}

File: app/src/main/java/com/example/myapplication/View/NotebookFragment.kt
package com.example.myapplication.View

import androidx.recyclerview.widget.RecyclerView
import android.view.LayoutInflater
import android.view.ViewGroup
import android.os.Bundle
import androidx.lifecycle.LifecycleOwner
import Models.linked.Product
import android.view.View
import androidx.fragment.app.Fragment
import androidx.lifecycle.Observer
import com.example.myapplication.ViewModel.NotebookViewModel
import com.example.myapplication.Adapter.CatalogAdapter
import com.example.myapplication.Other.VerticalSpacesItemDecoration
import com.example.myapplication.databinding.FragmentNotebookBinding
import org.koin.androidx.viewmodel.ext.android.viewModel
import java.util.*

class NotebookFragment : Fragment(), LifecycleOwner {
    private var binding: FragmentNotebookBinding? = null
    private val viewModel: NotebookViewModel by viewModel()
    var NotebookRecView: RecyclerView? = null
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?, savedInstanceState: Bundle?
    ): View? {
        binding = FragmentNotebookBinding.inflate(inflater, container, false)
        val root: View = binding!!.root
        NotebookRecView = binding!!.NotebookRecView

        val SpacesItemDecoration = VerticalSpacesItemDecoration(15)
        NotebookRecView!!.addItemDecoration(SpacesItemDecoration)
        viewModel.NotebookListLiveData.observe((context as LifecycleOwner?)!!, userListUpdateObserver)
        viewModel.GetNotebook(0, 5)
        return root
    }

    var userListUpdateObserver: Observer<ArrayList<Product>?> =
        Observer<ArrayList<Product>?> { userArrayList ->
            NotebookRecView!!.adapter = CatalogAdapter(userArrayList)
        }

    override fun onDestroyView() {
        super.onDestroyView()
        binding = null
    }
}